ci: bump kubernetes version in helm-ci

narqo opened the pull request:
#### What this PR does

The PR updates the Kubernetes version, that CI uses for the Helm chart tests from v1.20. ~~I picked v1.31 because it's the oldest version that is still supported.~~ v1.20 was released three years ago, and went out of support long since.

**Update** The CI uses kind v0.20, whose latest pre-build Kubernetes version was 1.29. I think it's good enough.

#### Checklist

- [x] Tests updated.
- [x] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`. If changelog entry is not needed, please add the `changelog-not-needed` label to the PR.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

dimitarvdimitrov reviewed (commented):
generally lgtm, but can you also modify where we mention what version of k8s we support that's mentioned in operations/helm/charts/mimir-distributed/README.md:23 and docs/sources/helm-charts/mimir-distributed/get-started-helm-charts/_index.md:32

narqo commented:
> modify where we mention what version of k8s

I've updated the chart's `Chart.yaml` to reflect the compatible version, and updated the docs. Also, added a changelog entry. PHAL

dimitarvdimitrov reviewed (approved):
LGTM, thanks
